app/controllers/settings/contact-info.js

```diff
@@ -0,0 +1,26 @@
+import Ember from 'ember';
+
+const { Controller } = Ember;
+
+export default Controller.extend({
+  actions: {
+    updateContactInfo() {
+      this.set('isLoading', true);
+      let currentUser = this.get('model');
+      currentUser.save()
+        .then(user => {
+          this.set('isLoading', false);
+          let userData = user.serialize(false).data.attributes;
+          userData.id = user.get('id');
+          this.get('session').set('data.currentUserFallback', userData);
+          this.get('notify', 'Updated information successfully');
+        })
+        .catch(error => {
+          this.set('isLoading', false);
+          this.get('authManager.currentUser').rollbackAttributes();
```

Comment:
Instead of doing it here, run `this.get('authManager.currentUser').rollbackAttributes();` when the user exits the route.